application-ideas-ui/src/main/resources/Ideas/IdeasClass.xml

```diff
@@ -539,80 +540,75 @@
         "$services.localization.render('ideas.ideasclass.vote.closed')", 'warning');
       return;
     }
+    if (docName != '') {
+      var docRef = XWiki.Model.resolve(docName, XWiki.EntityType.DOCUMENT, [xm.wiki]);
+      url = getRestURL(new XWiki.Document(docRef));
+    }
     var reflink = button.attr('class').split(' ')[1];
     var opp = button.data('opp');
     var refresh = button.data('ref');
-    // Creating the url according to the space from where the request was made
-    // and the type of action that is nedded.
+    // Determine the vote type that was cast
     if(voteClass == 'voted-pro') {
-      var addVote = 'addVote';
+      var addVote = true;
     } else {
-      var addVote = 'addVoteOpp';
+      var addVote = false;
     }
+    // Add the vote type to the url
+    url = url  + '?pro=' + addVote;
```

Comment:
If it did so, the data had to be accessed through the request body. And I was trying to avoid that on the server side. Anyway, I gave up on the query parameters and adopted another way of voting.